src/main/java/org/gnieh/logback/config/ConfigConfigurator.java

```diff
@@ -92,127 +93,83 @@ public void configure(LoggerContext loggerContext) {
-		PropertySetter propertySetter = new PropertySetter(beanCache, filter);
+	/**
+	 * Configure an object of a given class.
+	 * 
+	 * @param loggerContext
+	 *            the context to assign to this object if it is
+	 *            {@link ContextAwareBase}
+	 * @param clazz
+	 *            the class to instantiate
+	 * @param config
+	 *            a configuration containing the object's properties - each
+	 *            top-level key must have a corresponding setter method
+	 * @param children
+	 *            a list which, if not null, will be filled with any child
+	 *            objects assigned as properties
+	 * @return the object instantiated with all properties assigned
+	 * @throws ReflectiveOperationException
+	 *             if any setter method is missing or if the class cannot be
+	 *             instantiated with a no-argument constructor
+	 */
+	private <T> T configureObject(LoggerContext loggerContext, Class<T> clazz, Config config, List<Object> children) throws ReflectiveOperationException {
+		T object = clazz.newInstance();
+
+		if (object instanceof ContextAwareBase)
+			((ContextAwareBase) object).setContext(loggerContext);
+
+		PropertySetter propertySetter = new PropertySetter(beanCache, object);
 		for (Entry<String, ConfigValue> entry : config.withoutPath("class").root().entrySet()) {
 			String propertyName = NameUtils.toLowerCamelCase(entry.getKey());
-
-			Object value = entry.getValue().unwrapped();
-			if (value instanceof String) {
-				propertySetter.setProperty(propertyName, (String) value);
-			} else {
-				propertySetter.setComplexProperty(propertyName, value);
-			}
-		}
-
-		filter.setContext(loggerContext);
-		filter.start();
-
-		return filter;
-	}
-
-	private Encoder<?> configureEncoder(LoggerContext loggerContext, Config config)
-			throws ClassNotFoundException, InstantiationException, IllegalAccessException {
-		Class<?> clazz = Class.forName(config.getString("class"));
-		Encoder<?> encoder = (Encoder<?>) clazz.newInstance();
-
-		// for each key appearing in the encoder configuration, check whether
-		// it is a bean property and call the correct method on the encoder
-		PropertySetter propertySetter = new PropertySetter(beanCache, encoder);
-		for (Entry<String, ConfigValue> entry : config.withoutPath("class").root().entrySet()) {
-			String propertyName = NameUtils.toLowerCamelCase(entry.getKey());
-			if ("layout".equals(propertyName)) {
-				Layout<?> layout = configureLayout(loggerContext, config.getConfig("layout"));
-				propertySetter.setComplexProperty(propertyName, layout);
-			} else {
-				Object value = entry.getValue().unwrapped();
-				if (value instanceof String) {
-					propertySetter.setProperty(propertyName, (String) value);
+			ConfigValue value = entry.getValue();
+			switch (value.valueType()) {
+			case STRING:
+			case NUMBER:
+			case BOOLEAN:
+				propertySetter.setProperty(propertyName, config.getString(entry.getKey()));
```

Comment:
This looks weird, why not using numeric and boolean values? I wanted to write my own property setter to use the full power of typed values in the config library, maybe I should do so first, to ease this part.

Reply:
Yes, passing through `String` systematically for simple properties is strange. It's why at first I thought the `PropertySetter` didn't handle `ints` at all, until I looked more deeply into the code.

Reply:
Ok, I didn’t remember it worked this way. But are we sure that what is interpreted as a boolean by the config library will be by the property setter? (e.g. on/off, yes/no)

Reply:
No, but there's not much we can do about that without rewriting the PropertySetter or adding hacks here. I tried:
<pre>case BOOLEAN:
  propertySetter.setProperty(propertyName, Boolean.toString(config.getBoolean(entry.getKey())));
  break;
</pre>
But if you place "foo=yes" in the config, Config reports "foo" as having ValueType STRING. It will read it correctly if you try `getBoolean("foo")`, but will not report ValueType BOOLEAN unless you use true/false.

Reply:
Ok. I will add my own propertySetter implementation.

Reply:
See #3